Create lwm2m-agent-1020-440-39-realtime-session-resubscription-after-core-node-redirection.md (#3344)
SoftwareAG/c8y-docs: content/change-logs/device-management/lwm2m-agent-1020-440-39-realtime-session-resubscription-after-core-node-redirection.md

---

> date: ""

Reviewer: ```suggestion
date: 2025-05-22
```

Reviewer: This is the date for 2025.2, so we can add it already here

> title: Realtime Session Resubscription After Core Node Redirection

Reviewer: ```suggestion
title: Real-time session is properly resubscribed after core node redirection
```

> title: LWM2M device broken real-time operation subscriptions during Core upgrades

Reviewer: ```suggestion
title: LWM2M device real-time operation subscriptions no longer break during core upgrades
```

product_area: Device management & connectivity
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: component-1KLUzmqfe
    label: LWM2M
build_artifact:
  - value: tc-ggH2M4hf3
    label: lwm2m-agent
ticket: DM-4774
version: 1020.440.39
---

> When the Core platform is upgraded/restarted, in LWM2M Service this caused a side-effect of not having an active real-time subscription for operations anymore. It required a LWM2M new registration or operation subscription time to live duration to be expired to recover from this state.

Reviewer: ```suggestion
When the core platform was upgraded/restarted, this caused the LWM2M service no longer having an active real-time subscription for operations. It required a LWM2M new registration or the expiration of an operation subscription time-to-live duration to recover from this state.
```

> Previously, when a device's realtime subscription was marked as non-autoretriable and its session was redirected from one core node to another, the new core node did not inherit the existing subscription. As a result, the subscription was lost, and only a warning was logged without triggering a resubscription.  

Reviewer: ```suggestion
Previously, when a device's real-time subscription was marked as non-autoretryable and its session was redirected from one core node to another, the new core node did not inherit the existing subscription. As a result, the subscription was lost, and only a warning was logged without triggering a resubscription.  
```

Reviewer: non-autoretryable: This term is quite hard to read. Would it be possible to rephrase or is this a fix term?

> Previously, when a device's realtime subscription was marked as non-autoretriable and its session was redirected from one core node to another, the new core node did not inherit the existing subscription. As a result, the subscription was lost, and only a warning was logged without triggering a resubscription.  

Reviewer 2: To me this release note is too technical for a user. How about the following?
"
Title: LWM2M device broken real-time operation subscriptions during Core upgrades

Description: When the Core platform is upgraded/restarted, in LWM2M Service this caused a side-effect of not having an active real-time subscription for operations anymore. It required a LWM2M new registration or operation subscription time to live duration to be expired to recover from this state.
This issue has been resolved. Now, in such cases, the real-time session will be properly resubscribed on the new core node, ensuring continued data flow and improved reliability.
"

Reviewer 3: done

> This issue has been resolved. Now, in such cases, the realtime session will be properly resubscribed on the new core node, ensuring continued data flow and improved reliability.

Reviewer: ```suggestion
This issue has been resolved. Now, in such cases, the real-time session will be properly resubscribed on the new core node, ensuring continued data flow and improved reliability.
```
